src/ape/utils/abi.py

```diff
@@ -245,15 +249,27 @@ def is_dynamic_sized_type(abi_type: Union[ABIType, str]) -> bool:
     return parsed.is_dynamic
 
 
-class LogInputABICollection:
-    def __init__(self, abi: EventABI):
-        self.abi = abi
-        self.topic_abi_types = [i for i in abi.inputs if i.indexed]
-        self.data_abi_types: List[EventABIType] = [i for i in abi.inputs if not i.indexed]
+class LogInputABICollection(BaseModel):
+    abi: EventABI
+
+    class Config:
```

Comment:
I only suggest moving this to the bottom of the object. I looks better IMO. But it doesn't bother me either way

Reply:
I put it here for consistency's sake because all the other models in the codebase put it directly below the fields

Reply:
I put it here for consistency's sake because all the other models in the codebase put it directly below the fields